src/Connectors/src/CloudFoundry/CloudFoundryServiceInfoCreator.cs

```diff
@@ -12,50 +13,48 @@ namespace Steeltoe.Connector.CloudFoundry
     public class CloudFoundryServiceInfoCreator : ServiceInfoCreator
     {
         private static readonly object _lock = new object();
-        private static IConfiguration _config;
         private static CloudFoundryServiceInfoCreator _me = null;
 
-        internal CloudFoundryServiceInfoCreator(IConfiguration config)
-            : base(config)
+        internal CloudFoundryServiceInfoCreator(IConfiguration configuration)
+            : base(configuration)
         {
-#pragma warning disable S3010 // Static fields should not be updated in constructors
-            _config = config;
-#pragma warning restore S3010 // Static fields should not be updated in constructors
             BuildServiceInfoFactories();
             BuildServiceInfos();
         }
 
-        public static new CloudFoundryServiceInfoCreator Instance(IConfiguration config)
+        public static new CloudFoundryServiceInfoCreator Instance(IConfiguration configuration)
         {
-            if (config == null)
+            if (configuration == null)
             {
-                throw new ArgumentNullException(nameof(config));
+                throw new ArgumentNullException(nameof(configuration));
             }
 
-            if (config == _config)
+            if (configuration == _me?.Configuration)
```

Comment:
This section, from here to the final return can be simplified into a single return by converting the == into !=

Reply:
I think this return exists to avoid the (presumably negligible) perf hit from the lock. I think this return makes sense but inverting the logic below is also good

Reply:
To be clear, you still need to the `if` checks.  

I was suggesting:
```
if (configuration != _me?.Configuration)
{
    lock (_lock)
    {
        if (configuration != _me?.Configuration)
        {
            _me = new CloudFoundryServiceInfoCreator(configuration);
        }
    }
}
return _me;
```
From performance perpspective, 2 blocks identical.